src/include/sof/math/trig.h

```diff
@@ -137,17 +152,137 @@ static inline int16_t sin_fixed_16b(int32_t th_rad_fxp)
+		if (saturatedUnaryMinus <= INT32_MIN)
+			saturatedUnaryMinus = INT32_MAX;
+		else
+			saturatedUnaryMinus = -saturatedUnaryMinus;
+
+	    th_rad_fxp =
+		PI_MUL2_Q4_28 - iscalarcordic(saturatedUnaryMinus, CORDIC_31B_TABLE_SIZE,
+		cordic_ilookup, type);
+	}
+}
+
+/**
+ * CORDIC-based approximation of inverse cosine
+ * inverse cosine of cdc_acos_th based on a CORDIC approximation
+ * Inverse cosine angle values in rad
+ * Q2.30 cdc_acos_th , input value range [-1 to 1]
+ * Q2.30 th_rad_fxp, output value range [3.14159265346825 to 0]
+ * LUT size set type 15
+ * Error (max = 0.000000026077032), THD+N  = -157.948952635422842
+ */
+static inline void acosine_fixed_16b(int32_t cdc_acos_th, int32_t th_rad_fxp)
+{
+	int32_t saturatedUnaryMinus;
+	cordic_cfg type = EN_32B_CORDIC_ACOSINE;
+
+	saturatedUnaryMinus = cdc_acos_th;
+	if (saturatedUnaryMinus >= 0) {
+		th_rad_fxp = iscalarcordic(saturatedUnaryMinus, CORDIC_15B_ITABLE_SIZE,
+					   cordic_ilookup, type);
+	} else {
+		if (saturatedUnaryMinus <= INT32_MIN)
+			saturatedUnaryMinus = INT32_MAX;
+		else
+			saturatedUnaryMinus = -saturatedUnaryMinus;
+
+		th_rad_fxp =
+		PI_MUL2_Q4_28 - iscalarcordic(saturatedUnaryMinus, CORDIC_15B_ITABLE_SIZE,
+		cordic_ilookup, type);
+	}
+}
+
+/**
+ * CORDIC-based approximation of inverse sine
+ * inverse sine of cdc_asin_th based on a CORDIC approximation.
+ * Inverse sine angle values in rad
+ * Q2.30 cdc_asin_th,input value range [-1 to 1]
+ * Q2.30 th_rad_fxp, output value range [-1.5707963258028 to 1.5707963258028]
+ * LUT size set type 31
+ * Error (max = 0.000000027939677), THD+N  = -157.454534077921551
+ */
+static inline void asin_fixed_32b(int32_t cdc_asin_th, int32_t th_rad_fxp)
+{
+	int32_t saturatedUnaryMinus;
+	cordic_cfg type = EN_32B_CORDIC_ASINE;
+
+	saturatedUnaryMinus = cdc_asin_th;
+	if (saturatedUnaryMinus >= 0) {
+		th_rad_fxp = iscalarcordic(saturatedUnaryMinus, CORDIC_31B_TABLE_SIZE,
+					   cordic_ilookup, type);
+	} else {
+		if (saturatedUnaryMinus <= INT32_MIN)
+			saturatedUnaryMinus = INT32_MAX;
+		else
+			saturatedUnaryMinus = -saturatedUnaryMinus;
+
+	th_rad_fxp = -iscalarcordic(saturatedUnaryMinus, CORDIC_31B_TABLE_SIZE,
+				    cordic_ilookup, type);
+	}
+}
+
+/**
+ * CORDIC-based approximation of inverse sine
+ * inverse sine of cdc_asin_th based on a CORDIC approximation.
+ * Inverse sine angle values in rad
+ * Q2.30 cdc_asin_th, value in between range of [-1 to 1]
+ * Q2.30 th_rad_fxp, output value range [-1.5707963258028 to 1.5707963258028]
+ * LUT size set type 15
+ * Error (max = 0.000000027939677), THD+N  = -157.454534077921551
+ */
+static inline void asin_fixed_16b(int32_t cdc_asin_th, int32_t th_rad_fxp)
+{
+	int32_t saturatedUnaryMinus;
+	cordic_cfg type = EN_32B_CORDIC_ASINE;
+
+	saturatedUnaryMinus = cdc_asin_th;
+	if (saturatedUnaryMinus >= 0) {
+		th_rad_fxp = iscalarcordic(saturatedUnaryMinus, CORDIC_15B_ITABLE_SIZE,
+					   cordic_ilookup, type);
+	} else {
+		if (saturatedUnaryMinus <= INT32_MIN)
```

Comment:
is your sign right here? this var can never be less INT32_MIN, or should this be a int64?